tldw_Server_API/app/core/LLM_Calls/routing/rules_router.py

```
@@ -0,0 +1,33 @@
+"""Deterministic fallback routing for auto-model selection."""
+
+from __future__ import annotations
+
+from typing import Iterable, Mapping, Optional
```

Ruff (0.15.6)

[warning] Import from `collections.abc` instead: `Iterable`, `Mapping`

Import from `collections.abc`

(UP035)